pkgs/build-support/setup-hooks/separate-debug-info.sh

```diff
@@ -1,6 +1,13 @@
 export NIX_SET_BUILD_ID=1
-export NIX_LDFLAGS+=" --compress-debug-sections=zlib"
-export NIX_CFLAGS_COMPILE+=" -ggdb -Wa,--compress-debug-sections"
+export NIX_CFLAGS_COMPILE+=" -ggdb"
+if [[ -z "${dontCompressDebugSections-}" ]]; then
+    # Sometimes tools try to process debugging sections and fail on
+    # unusual compression mode like `bpftool`:
```

Comment:
Do we know of any other tools having problem with this as well, or only about bpftool?

Assuming bpftool gets fixed, I'd rather undo this additional flag, rather than keeping it around because "some tool might still need it"